### PR TITLE
hwmon: (pmbus/ltc2978) Add support for new sensor

### PR DESCRIPTION
**Pull request by msiddare**
Current ltc2978 driver does not have support for device ltc2979.

Added support for the sensor device ltc2979.

Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>

### CODE REVIEW

File: patch/cisco-add-support-for-ltc2979-chip.patch [hunk 1]
```diff
@@ -0,0 +1,74 @@
+From 81db6d9122f00751609393dc4265c205215c22df Mon Sep 17 00:00:00 2001
+From: msiddare <msiddare@contoso.com>
+Date: Mon, 7 Jun 2021 20:07:57 +0000
+Subject: [PATCH] hwmon: (pmbus/ltc2978) Add support for new sensor
+
+Current ltc2978 driver does not have support for device ltc2979.
+
+Added support for the sensor device ltc2979.
+
```
Comment (paulmenzel) [inline patch/cisco-add-support-for-ltc2979-chip.patch:9]: Please mention, what datasheet (name, revision) you used, and how it was tested.

Comment (msiddare) [reply patch/cisco-add-support-for-ltc2979-chip.patch:11]: Updated patch with data sheet of chip and Test results.

Please review

**Comment by paulmenzel** [comment]
Ha, you should have checked the Linus’ master branch first: [commit ee44fafb3886 (hwmon: (pmbus/ltc2978) add support for more parts)](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=ee44fafb38869e89ff1012134ea80d55e6e346fc) was added in Linux 5.7.

Please send a message to <stable@vger.kernel.org> asking for a backport to the stable series, and use that patch (`git format-patch -) instead of yours. Please put your testing data in the SONiC commit message (not the patch file in `patch`).

**Comment by lguohan** [comment]
@msiddare, please address

**Comment by anshuv-mfst** [comment]
@lguohan - to help with merge, thanks

File: patch/cisco-add-support-for-ltc2979-chip.patch [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
++	{ .compatible = "lltc,ltc2979" },
+ 	{ .compatible = "lltc,ltc2980" },
+ 	{ .compatible = "lltc,ltc3880" },
+ 	{ .compatible = "lltc,ltc3882" },
+@@ -788,5 +795,5 @@ static struct i2c_driver ltc2978_driver = {
+ module_i2c_driver(ltc2978_driver);
+ 
+ MODULE_AUTHOR("Guenter Roeck");
+-MODULE_DESCRIPTION("PMBus driver for LTC2978 and comppatible chips");
++MODULE_DESCRIPTION("PMBus driver for LTC2978 and compatible chips");
```
Comment (lguohan) [inline patch/cisco-add-support-for-ltc2979-chip.patch:140]: i do not see this change in the upstream patch, please remove.

**Comment by lguohan** [comment]
the upstream patch is available at https://github.com/torvalds/linux/commit/ee44fafb38869e89ff1012134ea80d55e6e346fc

File: patch/cisco-add-support-for-ltc2979-chip.patch (hunk 2 above)
Comment (msiddare) [reply patch/cisco-add-support-for-ltc2979-chip.patch:140]: Hi Guhon,

Upstream patch has corrected the spelling mistake in the above code. 
This is the line from  torvalds/linux@ee44faf 5.7-rc1
"MODULE_DESCRIPTION("PMBus driver for LTC2978 and compatible chips");"

Please let me know if you still want me to remove this change.

Comment (paulmenzel) [reply patch/cisco-add-support-for-ltc2979-chip.patch:140]: As discussed today, please create two patches with `git format-patch`:

    $ git format-patch -1 ee44fafb38869e89ff1012134ea80d55e6e346fc
    $ git format-patch -1 89688e8d1ecda59edc4e0ceb430aedaac18ca071

Comment (msiddare) [reply patch/cisco-add-support-for-ltc2979-chip.patch:140]: Patches created with with format-patch does not apply clean with 202012 branch kernel [ 4.19.152 ]

Please approve this patch for 202012 branch. 

Will remove this patch when migrating to 5.10 kernel

**Review by lguohan — APPROVED** [review]

File: patch/cisco-add-support-for-ltc2979-chip.patch (hunk 2 above)
Comment (paulmenzel) [reply patch/cisco-add-support-for-ltc2979-chip.patch:140]: > Patches created with with format-patch does not apply clean with 202012 branch kernel [ 4.19.152 ]

What other commits are missing to get it applied cleanly?

**Pull request closed without merging**